alarm/alarm.py

```diff
@@ -1,58 +1,160 @@
+    duration_group.add_argument(
+        "-s", "--seconds", type=int, help="Number of seconds before playing alarm"
+    )
+    duration_group.add_argument(
+        "-m", "--minutes", type=int, help="Number of minutes before playing alarm"
+    )
+
+    run_mode_group = parser.add_mutually_exclusive_group()
+    run_mode_group.add_argument(
+        "-b",
+        "--background",
+        action="store_true",
+        default=False,
+        help="Run timer in the background",
+    )
+    run_mode_group.add_argument(
+        "-d",
+        "--display_timer",
+        action="store_true",
+        default=False,
+        help="Show timer in console",
+    )
+
+    alarm_file_group = parser.add_mutually_exclusive_group()
+    alarm_file_group.add_argument(
+        "-l", "--song_library", help="Take a random song from a song library directory"
+    )
+    alarm_file_group.add_argument(
+        "-f", "--file", help="File path to song to play as alarm"
+    )
+    alarm_file_group.add_argument(
+        "-v", "--version", action="version", version=f"%(prog)s {version}"
+    )
+    return parser.parse_args(args)
+
+
+def _get_file(args) -> str:
+    """Get alarm song file from library, passed in file or environment"""
+    if args.song_library:
+        music_files = list(Path(args.song_library).rglob("*.mp[34]"))
+        if not music_files:
+            raise AlarmFileException(f"No music files found in {args.song_library}")
+        return str(random.choice(music_files))
+    elif args.file:
+        return args.file
+    else:
+        return os.environ["ALARM_MUSIC_FILE"]
+
+
+def _validate_file(file: str) -> None:
+    """Make sure we get a music file that exists"""
+    if not Path(file).exists():
+        raise AlarmFileException(f"{file} does not exist")
+    allowed_extensions = ("mp3", "mp4", "wav")
+    if not Path(file).suffix.endswith(allowed_extensions):
+        raise AlarmFileException(
+            f"{file} is not supported ({', '.join(allowed_extensions)} files are)"
+        )
 
 
-if __name__ == '__main__':
-    if len(sys.argv) > 1:
-        minutes = int(sys.argv[1])
-        countdown(minutes * 60)
+def get_alarm_file(args) -> str:
+    file = _get_file(args)
+    _validate_file(file)
+    return file
+
+
+def main(args=None):
+    args = args if args else parse_args(sys.argv[1:])
+
+    if args.seconds:
+        seconds = int(args.seconds)
+        minutes = seconds / 60
+    else:
+        minutes = int(args.minutes)
+        seconds = minutes * 60
+
+    alarm_file = get_alarm_file(args)
+
+    if args.background:
+        if minutes < 1:
+            time_till_alarm = f"{seconds} seconds"
+        else:
+            time_till_alarm = f"{minutes} minute{'' if minutes == 1 else 's'}"
+
+        print(f"Playing alarm in {time_till_alarm}")
+
+        package = __package__
+        module = Path(sys.argv[0]).stem
+
+        os.system(f"python -m {package}.{module} -s {seconds} -f '{alarm_file}' &")
```

Comment:
Ah very nice, this explains my question about args I guess, but still this would be covered by args, right? But interesting that you can call a module via package.module directly without __main__.py?

Reply:
yeah seems so and without __main__.py :)
thanks for that background process suggestion, makes this tool so much more valuable